src/core/cgroup.c

```diff
@@ -3342,8 +3345,14 @@ int manager_setup_cgroup(Manager *m) {
                 return log_error_errno(r, "Failed to disable cgroup empty event source: %m");
 
         (void) sd_event_source_set_description(m->cgroup_empty_event_source, "cgroup-empty");
+        return 0;
+}
+
+static int manager_setup_cgroup_location_and_empty_notifications(Manager *m) {
```

Comment:
that's quite a function name! 

why not split this into two functions? it appears to me the empty notification stuff and the creation of cgroup for pid1 are actually quite distinct pieces of code for which two separate functions would be quite befitting